vncdotool/loggingproxy.py

```diff
@@ -31,19 +32,20 @@ def connectionMade(self):
         Protocol.connectionMade(self)
         self.transport.setTcpNoDelay(True)
 
-        self.buffer = ''
+        self.buffer = b''
         self.nbytes = 0
         # XXX send version message
         self._handler = self._handle_version, 12
 
     def dataReceived(self, data):
-        self.buffer += data
+        self.buffer = data + self.buffer
```

Comment:
These aren't equivalent? I'm not sure how this worked at all unless the buffer was always empty when you received new data, as they'll put it in the wrong order?

Reply:
That was a mistake, i'm glad you spotted it.
Must have been an empty buffer as you point out.

Reply:
Do note that compound assignment of bytes objects is not possible in python 3.